src/Tokenizer/Tokenizer.php

```diff
@@ -269,10 +269,26 @@ private function tokenizeMultilineComment(
         }
 
         if ($matches = $line->peek(self::TOKEN_COMMENT_MULTILINE_END)) {
-            $token = $state->endMultilineToken($matches[0]);
+
+            $values = [''];
+            foreach ($tokens->build() as $commentLine) {
+                $values[] = $commentLine->getValue();
+            }
+            $values[] = $matches[0];
+
+            $token = $state->endMultilineToken(implode("\n", $values));
+
             $tokens->appendToken($token);
             return;
         }
+
+        $token = new Token(
+            TokenInterface::TYPE_COMMENT_LINE,
+            $line->value(),
+            $line->index()
+        );
+
+        $tokens->appendToken($token);
```

Comment:
OK, after a bit of agonizing over this, I finally got it. It looks like the `COMMENT_LINE` token is used as kind of a... helper token (?) to buffer individual comment lines before they are all concatenated into the final `COMMENT_MULTILINE` token when the `*/` sequence is encountered.

This would leave redundant tokens in the token stream (which is probably the reason that they're filtered out again before parsing, in the `TokenStream` class). It would probably also break the various implementations of the `TokenPrinterInterface` (like `CodeTokenPrinter`, which should reproduce the exact original code from a token stream -- actually, we could easily use the existing test fixtures to add test cases for that class... 🤔).

Actually, this is exactly what the `MultilineTokenBuilder` (here stored in `$state`) is for. So actually, this might actually suffice (⚠️ entirely untested 💀):

```suggestion
            $token = $state->endMultilineToken($matches[0]);
            $tokens->appendToken($token);
            return;
        }

        $state->appendToToken($line->value());
```